implementation/query-runner/src/main/java/net/opentsdb/tsd/QueryConfig.java

```diff
@@ -263,14 +277,14 @@ public void cancel() {
   class ResponseCallback implements FutureCallback<HttpResponse> {
     private final int index;
     private final TSDB tsdb;
-    private StatsTimer timer;
+    private volatile StatsTimer timer;
```

Comment:
We need timer to be volatile? I see it being used in the synchronized methods only.

Reply:
Yeah I can yank that now, thanks.